modules/miner/splitsetheap_test.go

```diff
@@ -0,0 +1,450 @@
+				size:         uint64(10 * i),
+				transactions: make([]types.Transaction, 0),
+			},
+
+			id:    splitSetID(i),
+			index: 0,
+		}
+		e2 := &mapElement{
+			set: &splitSet{
+				averageFee:   types.SiacoinPrecision.Mul64(uint64(i)),
+				size:         uint64(10 * i),
+				transactions: make([]types.Transaction, 0),
+			},
+
+			id:    splitSetID(i),
+			index: 0,
+		}
+		max.Push(e1)
+		min.Push(e2)
+	}
+
+	for i := 0; i < 100; i++ {
+		maxPop := max.Pop()
+		minPop := min.Pop()
+
+		if int(maxPop.id) != 99-i {
+			t.Error("Unexpected splitSetID in result from max-heap pop.")
+		}
+
+		if int(minPop.id) != i {
+			t.Error("Unexpected splitSetID in result from min-heap pop.")
+		}
+
+		if maxPop.set.averageFee.Cmp(types.SiacoinPrecision.Mul64(uint64(99-i))) != 0 {
+			t.Error("Unexpected currency value in result from max-heap pop.")
+		}
+
+		if minPop.set.averageFee.Cmp(types.SiacoinPrecision.Mul64(uint64(i))) != 0 {
+			t.Error("Unexpected currency value in result from min-heap pop.")
+		}
+	}
+}
+
+// TestMapHeapSize tests that the size of MapHeaps changes accordingly with the
+// sizes of elements added to it, and with those elements removed from it.
+// Tests a max-heap and min-heap on the same sequence of pushes and pops.
+func TestMapHeapSize(t *testing.T) {
+	max := &mapHeap{
+		selectID: make(map[splitSetID]*mapElement),
+		data:     make([]*mapElement, 0),
+		size:     0,
+		minHeap:  false,
+	}
+
+	min := &mapHeap{
+		selectID: make(map[splitSetID]*mapElement),
+		data:     make([]*mapElement, 0),
+		size:     0,
+		minHeap:  true,
+	}
+
+	max.Init()
+	min.Init()
+
+	var expectedSize uint64
+
+	for i := 0; i < 1000; i++ {
+		e1 := &mapElement{
+			set: &splitSet{
+				averageFee:   types.SiacoinPrecision.Mul64(uint64(i)),
+				size:         uint64(100 * i),
+				transactions: make([]types.Transaction, 0),
+			},
+
+			id:    splitSetID(i),
+			index: 0,
+		}
+		e2 := &mapElement{
+			set: &splitSet{
+				averageFee:   types.SiacoinPrecision.Mul64(uint64(i)),
+				size:         uint64(100 * i),
+				transactions: make([]types.Transaction, 0),
+			},
+
+			id:    splitSetID(i),
+			index: 0,
+		}
+		max.Push(e1)
+		min.Push(e2)
+		expectedSize += e1.set.size
+	}
+
+	if max.size != expectedSize {
+		t.Error("Max-heap size different than expected size.")
+	}
+	if min.size != expectedSize {
+		t.Error("Min-heap size different than expected size.")
+	}
+
+	for i := 0; i < 1000; i++ {
+		maxPop := max.Pop()
+		minPop := min.Pop()
+
+		if maxPop.set.size != uint64(100*(999-i)) {
+			t.Log(i)
+			t.Error("Unexpected set size in result from max-heap pop.")
+		}
+
+		if minPop.set.size != uint64(100*i) {
+			t.Log(i)
+			t.Error("Unexpected set size in result from min-heap pop.")
+		}
+
+	}
+}
+
+// TestMapHeapRemoveBySetID pushes a sequence of elements onto
+// a max-heap and min-heap. Then it removes a random element using
+// its splitSetID, and checks that it has been removed.
+func TestMapHeapRemoveBySetID(t *testing.T) {
```

Comment:
Looks like you do not check that the size of the heap stays consistent when removing by id. Need to be testing for all of the stateful conditions of the heap when testing specific operations.